File: Daily Test/Java/Position Product 1.7.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[n];
        
        for(int i = 0; i < n; i++)
            a[i] = sc.nextInt();
        
        for(int i = 0; i < n; i++) {
            int idx = (a[i] > 0) ? a[i] - 1 : a[i];
            System.out.print(a[i] * a[idx] + " ");
        }
    }
}

File: Daily Test/C/Position Product 1.7.2025.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);
    int a[n];
    
    for(int i = 0; i < n; i++)
        scanf("%d", &a[i]);
        
    for(int i = 0; i < n; i++) {
        int idx = (a[i] > 0) ? (a[i] - 1) : a[i];
        printf("%d ", a[i] * a[idx]);
    }
    
    return 0;
}